subprojects/packagefiles/cmark-gfm/meson.build

```diff
@@ -0,0 +1,123 @@
+project(
+    'cmark-gfm',
+    'c',
+    version: '0.29.0.gfm.6',
+    default_options: ['warning_level=3'],
+    meson_version: '>=0.49.0',
+    license : 'BSD-2-Clause',
+)
+
+cc = meson.get_compiler('c')
+has_stdbool_header = cc.has_header('stdbool.h')
+has_builtin_expect_code = 'int main() { __builtin_expect(0,0); return 0; }'
+has_builtin_expect = cc.compiles(
+    has_builtin_expect_code,
+    name: 'has __builtin_expect',
+)
+has_attribute_code = '''
+  int f(void) __attribute__ (());
+  int main() { return 0; }
+'''
+has_attribute = cc.links(has_attribute_code, name: 'has __attribute__')
+
+cdata = configuration_data(
+    {
+        'HAVE_STDBOOL_H': has_stdbool_header,
+        'HAVE___BUILTIN_EXPECT': has_builtin_expect,
+        'HAVE___ATTRIBUTE__': has_attribute,
+    },
+)
+
+configure_file(
+    format: 'cmake@',
+    input: 'src/config.h.in',
+    output: 'config.h',
+    configuration: cdata,
+)
+
+varr = meson.project_version().split('.')
+vdata = configuration_data(
+    {
+        'PROJECT_VERSION_MAJOR': varr[0],
+        'PROJECT_VERSION_MINOR': varr[1],
+        'PROJECT_VERSION_PATCH': varr[2],
+        'PROJECT_VERSION_GFM': varr[4],
+    },
+)
+
+configure_file(
+    format: 'cmake@',
+    input: 'src/cmark-gfm_version.h.in',
+    output: 'cmark-gfm_version.h',
+    configuration: vdata,
+)
+
+edata = configuration_data(
+    {
+        'CMARK_GFM_STATIC_DEFINE': get_option('default_library') == 'static',
```

Comment:
(Keeping in mind this is only needed for Windows, i.e. when `_WIN32 || __CYGWIN__` is defined, in the dllexport code branch.)

We should probably also add this define as a cflag for the declare_dependency and the pkg-config file, if it ends up being true and needed on Windows for linkage.

Reply:
Looks like [upstream defines them as cflags](https://github.com/github/cmark-gfm/blob/master/extensions/CMakeLists.txt#L54-L56) too, I've made a commit to match that behavior. As for the pkg-config, I'm not sure I follow. Do we need to define one? Likewise the Windows linker errors are a mystery to me.

Reply:
I was going to say they were needed as compile_args in the declare_dependency, but...

I realized this is likely not true. For static builds, we should be covered by:
```
/* Is this an exclusively static build */
#if @CMARK_GFM_STATIC_DEFINE@ && ! defined CMARK_GFM_STATIC_DEFINE
#  define CMARK_GFM_STATIC_DEFINE
#endif
```

Reply:
Do we revert https://github.com/mesonbuild/wrapdb/pull/719/commits/36fd79c33555afa7e3c0b76637ea0cd515bf1b4a then?

Reply:
Indeed I believe that is not necessary either.

Reply:
Reverted it.